### PR TITLE
[FrameworkBundle] Display aliases in `debug:container` command

### PR DESCRIPTION
**Pull request by 94noni**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | Restart https://github.com/symfony/symfony/pull/47427
| License       | MIT

---

> [!NOTE]
> This PR restarts the work achieved by @jlslew in https://github.com/symfony/symfony/pull/47427
> as per https://github.com/symfony/symfony/pull/47427#issuecomment-2003011062

I do think this PR can give some valueable data ([like one of my prev PR](https://github.com/symfony/symfony/pull/46814))
so I am continuing the work already done, to push it forward :) 

I may need help/guidance for understanding the tests logic of such descriptor, as they are quite complicated :s 

---

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php [hunk 1]
```diff
@@ -434,10 +435,20 @@ private function describeValue($value, bool $omitTags, bool $showArguments, ?Con
...
-            ];
+            $node = $container?->getCompiler()->getServiceReferenceGraph()->getNode((string) $value);
+
+            if ($withAliases && $node?->getValue()?->getClass()) {
+                return [
+                    'type' => 'service',
+                    'id' => (string) $value,
+                    'alias' => $node->getValue()->getClass(),
+                ];
+            } else {
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php:446]: no need for the "else"

**Comment by nicolas-grekas** [comment]
Arf, I'm sorry but looking at the implementation, this is a false promise. This doesn't show aliases but only the class of the source service. This works for services that use the id=fqcn convention, but that's inaccurate in the general case.

`ContainerBuilderDebugDumpPass` runs `BEFORE_REMOVING` which means `ResolveReferencesToAliasesPass` already ran and we lost the information that would be needed to achieve this.

I don't know if we can reorder the passes to dump before resolving, but as is, this PR doesn't make sense to me.

**Comment by stof** [comment]
If we dump before resolving, we might end up missing other information in the debug commands by displaying the state before most compiler pass have run

**Comment by 94noni** [comment]
@nicolas-grekas oki i got you, to be fair I've juste restarted it, blindly copy/pasting for now so the PR get visible
if, for actual code in container, its not possible lets close as wontfix
thx

**Pull request closed without merging**